src/platform/forms-system/src/js/components/FieldTemplate.jsx

```diff
@@ -85,20 +77,20 @@ export default function FieldTemplate(props) {
   const labelElement = useFieldsetLegend ? (
     <legend id={`${id}-label`} className={labelClassNames}>
       {label}
-      {requiredSpan}
+      <RequiredSpan show={required} />
     </legend>
   ) : (
     <label id={`${id}-label`} className={labelClassNames} htmlFor={id}>
       {label}
-      {requiredSpan}
+      <RequiredSpan show={required} />
     </label>
   );
 
   // Don't render hidden or empty labels - prevents duplicate IDs on review &
   // submit page
   const showLabel =
     !uiSchema['ui:options']?.hideLabelText &&
-    (typeof label !== 'string' || (requiredSpan || label.trim()));
+    (typeof label !== 'string' || (required || label.trim()));
```

Comment:
`requiredSpan` was truthy if `required` was truthy, and it makes more sense to check `required` here anyway.